src/main/java/com/books/services/BookService.java

```diff
@@ -60,17 +72,34 @@ public void addBook(Book book) {
         repository.add(book);
     }
 
-    public void setBook(int id, Book book) {
-        repository.setItem(id, book);
+    public void setBook(Book book) {
```

Comment:
It's unclear for me, what this method do. It seems like it's just a setter of singleton.